mesa/agent.py

```diff
@@ -41,12 +42,15 @@ def __init__(self, unique_id: int, model: Model) -> None:
         self.model = model
         self.pos: Position | None = None
 
-        # Register the agent with the model using defaultdict
-        self.model.agents[type(self)][self] = None
+        # Directly register the agent with the model
+        if type(self) not in self.model._agents:
+            self.model._agents[type(self)] = set()
```

Comment:
this should be a dict, right?

Reply:
I have focussed on AgentSet so far, but I will take a look at this next.